crates/ra_hir_def/src/item_scope.rs

```diff
@@ -30,7 +33,7 @@ pub struct ItemScope {
 static BUILTIN_SCOPE: Lazy<FxHashMap<Name, PerNs>> = Lazy::new(|| {
     BuiltinType::ALL
         .iter()
-        .map(|(name, ty)| (name.clone(), PerNs::types(ty.clone().into())))
+        .map(|(name, ty)| (name.clone(), PerNs::types(ty.clone().into(), Visibility::Public)))
```

Comment:
Hm, looks like builtins should have some other kind of visiblity, as you can't `use super::u32`, unless you also `use u32 as u32` in a parent module. But this probably doesn't matter outside of some extreme edge cases .

Reply:
I think that just needs to be handled in path resolution (i.e. we shouldn't consider the builtin scope when looking at a path with more than one segment? not sure if we currently do)

Reply:
Yeah, more generally, I think that treating builtins as just a separate scope was a mistake, we should rather special-case them in path resolution and in completions.